scipy/stats/_multivariate.py

```diff
@@ -2620,3 +2621,295 @@ def entropy(self):
     method_frozen.__doc__ = doccer.docformat(
         method.__doc__, wishart_docdict_noparams)
     method.__doc__ = doccer.docformat(method.__doc__, wishart_docdict_params)
+
+
+_doc_multinom_callparam = """\
+n : int
+    Number of trials
+p : array_like, shape (k,)
+    Probabilities. These should sum to one. If they do not, then
+    ``p[-1]`` is modified to account for the remaining probability so that
+    ``sum(p) == 1``.
+"""
+
+multinom_docdict_params = {
+    '_doc_multinom_callparam': _doc_multinom_callparam,
+    '_doc_random_state': _doc_random_state
+}
+
+class multinomial_gen(multi_rv_generic):
+    r"""Multinomial distribution.
+
+    The distribution is parameterized by an integer :math:`n` and a
+    k-dimensional vector of probabilities :math:`p`.
+
+    The support of the distribution is a set of k-dimensional vectors of
+    integer values :math:`\mathbf{x}`, such that :math:`\sum_{1}^{k} x_k = n`.
+
+    The probability mass function takes the form:
+
+    .. math::
+
+        P(\mathbf{x}) = \frac{n!}{x_1! \cdots x_{k}!} p_1^{x_1} \cdots p_k^{x_k}
+
+    .. versionadded:: 0.17
+
+    Parameters
+    ----------
+    n : int
+        Number of trials
+    p : array_like, shape (k,)
+        Probabilities. These should sum to one. If they do not, then
+        ``p[-1]`` is modified to account for the remaining probability so that
+        ``sum(p) == 1``.
+
+    %(_doc_random_state)s
+
+    Methods
+    -------
+    ``pmf(x, n, p)``
+    ``logpmf(x, n, p)``
+    ``rvs(n, p, size=1, random_state=None)``
+    ``mean(n, p)``
+    ``cov(n, p)``
+
+    Examples
+    --------
+
+    Draw random numbers from a multinomial distribution:
+
+    >>> n, p = 4, [0.3, 0.4, 0.4]
```

Comment:
As far as I got it, the potential problem is that re-normalizing might be harmful in cases when probabilities were already normalized (in some magically precise way). But is it really so? We need to simply add up values (not their squares, etc.), all of them lie in `[0, 1]`. I doubt it can cause serious numerical issues, but I'm not expert at all. 

Anyway, I would suggest the algorithm:

```
s = np.sum(p)
if np.abs(s - 1) > np.finfo(np.float).eps:
  warn('Probabilities do not add up to 1, they will be normalized.')
  p = p / s
```